### PR TITLE
Form field error size theme doc

### PR DESCRIPTION
**Pull request by g4rry420**
#### What does this PR do?
Added the docs for the FormField.error.size

#### Where should the reviewer start?
components => FormField => doc.js

#### What testing has been done on this PR?
I had run the yarn test-update command to update the snapshots

#### How should this be manually tested?
N/A

#### Any background context you want to provide?
N/A

#### What are the relevant issues?
N/A

#### Screenshots (if appropriate)
N/A

#### Do the grommet docs need to be updated?
No

#### Should this PR be mentioned in the release notes?
N/A

#### Is this change backwards compatible or is it a breaking change?
N/A

### CODE REVIEW

**Comment by g4rry420** [comment]
#4964

**Comment by g4rry420** [comment]
I got a question. How can we specify the other value of size property to users such as xsmall, small, large and xlarge ?

**Comment by ShimiSun** [comment]
> I got a question. How can we specify the other value of size property to users such as xsmall, small, large and xlarge ?

Good question, you can add the theme docs for `formField.error.size.small`, `formField.error.size.medium`(...) which will specify the actual pixel value for each.

**Comment by g4rry420** [comment]
How about if we do the same as in the global.borderSize in the formField ?

**Comment by ShimiSun** [comment]
> How about if we do the same as in the global.borderSize in the formField?

Sure. Thanks. It should already exist on the doc, might be worth moving it to docs utils and reuse it.

**Comment by g4rry420** [comment]
I am trying to use global.borderSize but somehow it's not working. Can you please tell how to use them ?

**Comment by ShimiSun** [comment]
```
  'global.borderSize': {
    description: 'The size of the border.',
    type: 'string',
    defaultValue: `{
      xsmall: '1px',
      small: '2px',
      medium: '4px',
      large: '12px',
      xlarge: '24px,
    }`,
  },
```

Taken from Tabs/doc.js

**Comment by g4rry420** [comment]
https://codesandbox.io/s/strange-newton-nlizl?file=/src/App.js

Some borderSize doesn't seem to work.

**Comment by ShimiSun** [comment]
I'm not sure what you are asking, on your sandbox you are using `formField.error.size` which is a prop that isn't mentioned on the docs and not something that grommet supports, so what were you expecting to see?
Also, `borderSize` is just the definition of sizes and it isn't stating default values. This definition is used in the background by grommet to look up what are the values of small/medium/large should be mapped to, e.g. `Box border={{size="large"...}}` would map to border size large according to the large value on `borderSize`.

**Comment by g4rry420** [comment]
ohh, okay, Thank You for the clarification. Btw, `formField.error.size ` in the theme does seems to work if I add different values such as xsmall, small, medium , large and xlarge.

**Comment by raywong-fb** [comment]
> ohh, okay, Thank You for the clarification. Btw, `formField.error.size ` in the theme does seems to work if I add different values such as xsmall, small, medium , large and xlarge.

Yes, `formField.error.size` does work as stated in my [original issue](https://github.com/grommet/grommet/issues/4964). The request was to simply add it to the documentation since it is a working spec.

File: src/js/components/FormField/doc.js [hunk 1]
```diff
@@ -175,6 +175,18 @@ export const themeDoc = {
...
+     The default size is medium.`,
+    type: 'string',
+    defaultValue: `
+      xsmall: '12px'
+      small: '14px'
+      medium: '18px'
+      large: '22px'
+      xlarge: '26px
+    `,
+  },
```
Comment (ShimiSun) [inline src/js/components/FormField/doc.js:189]:
```suggestion
  'formField.error.size': {
    description: `The size of the error messsage to be displayed.
     The default size is medium.`,
    type: 'string',
    defaultValue: medium',
  },
  'formField.error.size.small': {
    description: `The size of the error messsage to be displayed.
     The default size is medium.`,
    type: 'string',
    defaultValue: '12px',
  },
    'formField.error.size.medium': {
    description: `The size of the error messsage to be displayed.
     The default size is medium.`,
    type: 'string',
    defaultValue: '18px',
  },
  // same for the rest of the sizes.
```
It is a little confusing when we say the type of the prop is a  string, but showing an example of an object, the above is a cleaner way to suggest the changes.

**Review by ShimiSun — COMMENTED** [review]
Let me know if the approach I suggested makes sense?

File: src/js/components/FormField/doc.js (hunk 1 above)
Comment (g4rry420) [reply src/js/components/FormField/doc.js:189]: yes, this make more sense, shall I do that for the rest of it too ?

**Comment by g4rry420** [comment]
I don't understand why some tests are failing.

File: src/js/components/FormField/doc.js [hunk 2]
```diff
@@ -175,6 +175,42 @@ export const themeDoc = {
...
   },
+  'formField.error.size': {
+    description: `The size of the error messsage to be displayed.
+     The default size is medium.`,
+    type: 'string',
+    defaultValue: 'medium',
+  },
+  'formField.error.size.xsmall': {
+    description: `The size of the error messsage to be displayed.
+     The default size is medium.`,
```
Comment (ShimiSun) [inline src/js/components/FormField/doc.js:186]:
```suggestion
    description: `The size of a 'xsmall' error message.`,
```
The same fix for description should be applied to the rest of the sizes.
Also, there is a type for  'messsage' that should be 'message' instead across the other sizes..

**Review by ShimiSun — COMMENTED** [review]
Almost there! dropped a  comment.

File: src/js/components/FormField/doc.js (hunk 2 above)
Comment (g4rry420) [reply src/js/components/FormField/doc.js]: All done, Thank You for all your help and suggestions.

**Review by ShimiSun — APPROVED** [review]
Thank you for your contribution!

**Pull request merged**